src/components/map-controls.js

```diff
@@ -122,79 +143,98 @@ export default class MapControls extends PureComponent {
     ));
   }
 
-  _onTouchRotate(opts) {
-    this._onMouseRotate(opts);
-  }
-
-  _onMouseDown({pos}) {
+  _onDragStart(event) {
+    const pos = this._getCenter(event);
     const newMapState = this.props.mapState.panStart({pos}).rotateStart({pos});
     this._updateViewport(newMapState, {isDragging: true});
   }
 
-  _onMouseDrag({pos, startPos, modifier}) {
-    if (!this.props.onChangeViewport) {
-      return;
-    }
-
-    if (modifier) {
-      this._onMouseRotate({pos, startPos});
+  _onDrag(event) {
+    if (this._isFunctionKeyPressed(event)) {
```

Comment:
`(this._isFunctionKeyPressed(event) ? this._onRotate : this._onPan)(event);` FTW ;)